kotlinx-coroutines-test/common/src/internal/ReportingSupervisorJob.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2016-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.test.internal
+
+import kotlinx.coroutines.*
+
+/**
+ * A variant of [SupervisorJob] that additionally notifies about child failures via a callback.
+ */
+@Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE", "CANNOT_OVERRIDE_INVISIBLE_MEMBER")
+internal class ReportingSupervisorJob(parent: Job? = null, val onChildCancellation: (Throwable) -> Unit) :
+    JobImpl(parent) {
+    override fun childCancelled(cause: Throwable): Boolean = onChildCancellation(cause).let { false }
```

Comment:
It's potato-potato of course, but personally I find multi-liners here much more straightforward:
```
onChildCancellation(cause)
return false
```

Reply:
I like the one-liner approach because the exit points are self-evident. In the example you commented on, yes, this can go both ways. The code was modified and has more complex control flow, but the intent to always return `false` didn't change, and here the benefit, if there is any, should be more clear: the code clearly signifies that, no matter what happens, `false` should always be returned here. One doesn't have to scan the function for other possible exit points.